### DISCUSSION TITLE
How to specify multiple Consul cluster hostname

### DISCUSSION THREAD

**Discussion by liangrui1988** [cortexproject/cortex#5522] [Q&A] [answered]
-ring.store=consul \
 -consul.hostname=consul:8500 \

How to specify multiple Consul cluster hostname

I need ?
 -consul.hostname=consul1:8500,consul2:8500,consu3:8500 \

`level=error ts=2023-08-18T09:50:53.398938448Z caller=client.go:155 msg="error getting key" key=collectors/ring err="Get \"http://cortex-64-66.xx.com:8500,cortex-64-64.xx.com:8500,cortex-65-148.xx.com:8500/v1/kv/collectors/ring\": dial tcp: lookup cortex-64-66.xx.com:8500,cortex-64-64.xx.com:8500,cortex-65-148.xx.com: no such host"`

**Comment by friedrich-at-adobe** [comment, marked as answer]
if you use kubernetes you can use a service that points to those hosts, if you don't use kubernetes you need to define those hosts as one dns or have a loadbalancer in front of them
